### PR TITLE
postgresql_select: prepare for postgresql 17

### PR DESCRIPTION
**Pull request by dgilman**
#### Description

Update postgresql_select for postgresql 17. This will fail to build because many of the old postgresql versions do not build anymore. I wasn't able to remove them in time for the postgresql 17 release but they will hopefully be out shortly.

###### Type(s)

- [ ] bugfix
- x ] enhancement
- [ ] security fix

###### Tested on
macOS 14.6.1 23G93 x86_64
Xcode 16.0 16A242d

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL in commit message? 
- [x] checked your Portfile with `port lint --nitpick`?
- [x] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?
- [x] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

**Comment by ryandesign** [comment]
Select files get installed with the port. If you change something that a port installs, you have to increase its revision, otherwise existing users won't get the updated file.

**Comment by dgilman** [comment]
There are no new files for any of these versions of postgresql

**Comment by jmroot** [comment]
Ryan is correct, you are modifying the select file that is installed by each postgres port and this requires a revision bump.

To avoid the need to rebuild all the postgres versions just for a change to a select file in the future, you may like to change to the model used by python3_select, where each python3X port depends on python3_select-3X which is a subport of python3_select and installs the version-specific select file.

**Comment by reneeotten** [comment]
@dgilman can you please follow @jmroot his advice to avoid rebuilds in the future and make the suggested changes here instead/in addition to the `revision` bump?

**Comment by holymonson** [comment]
> To avoid the need to rebuild all the postgres versions just for a change to a select file in the future, you may like to change to the model used by python3_select, where each python3X port depends on python3_select-3X which is a subport of python3_select and installs the version-specific select file.

This is a common issue for all select-enhanced packages. In practice, those linking items are hardly changed between major versions, do we realy want many `XXX_select-YYY` subports to crowd registry, rather than a bulk of revision bump in rare modified cases?

**Comment by jmroot** [comment]
> In practice, those linking items are hardly changed between major versions, do we realy want many `XXX_select-YYY` subports to crowd registry, rather than a bulk of revision bump in rare modified cases?

Yes, we do. The actual cost of a few extra subports is extremely small.

**Comment by holymonson** [comment]
Checks failed for building legacy postgresql 84~94, could we get this done?

**Comment by dgilman** [comment]
This has been good to merge since day 1. The older versions are a known fail because they no longer build for API compatibility reasons. I have spent a few months getting the tree ready to delete the old postgresqls, however, it’s been a substantial project and I have not yet been able to delete any of the old versions.

**Comment by holymonson** [comment]
Maybe we should add `known_fail yes` into deprecated PG so those deprecated ports won't fail tests.

File: databases/postgresql10/Portfile [hunk 1]
```diff
@@ -1,21 +1,20 @@
 # -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
 
 PortSystem 1.0
-PortGroup select 1.0
```
Comment (reneeotten) [inline databases/postgresql10/Portfile:4]: If you remove the "select" PortGroup then if you remove the "select" PortGroup you also need to remove the runtime dependency on the "postgresql_select-CY" port and the actual select files in the "files" directory.

**Review by reneeotten — COMMENTED** [review]
You have added "know _fail" to many, many ports now... Is that only to satisfy the CI (then don't) or do they really not build on any OS anymore, which I kind of doubt.

**Comment by holymonson** [comment]
We want only postgresql 17, or furthermore only maintained branches, while those many deprecated postgresql branches we rather remain them untouched. But now asking `postgresql_select` to be compatible with ALL the branches while nobody really cares those deprecated causes it hard to upgrade.
I doubt if we need those deprecated postgresql before postgresql12, if someone does, IMHO, it would be better keep them untouched even it breaks `postgresql_select`.

**Comment by reneeotten** [comment]
> We want only postgresql 17, or furthermore only maintained branches, while those many deprecated postgresql branches we rather remain them untouched. But now asking `postgresql_select` to be compatible with ALL the branches while nobody really cares those deprecated causes it hard to upgrade. I doubt if we need those deprecated postgresql before postgresql12, if someone does, IMHO, it would be better keep them untouched even it breaks `postgresql_select`.

Nobody is asking that; things should be in a consistent state. I am sure @dgilman understands my comments. I am not opposed at all to merging this, but it has to be correct.

**Comment by dgilman** [comment]
IIRC there are a decent number of patches with fixes to API compat with openssl and libxml that haven't been backported to the old versions. Anything before 10 or 11 is totally written off at this point.

File: databases/postgresql10/Portfile (hunk 1 above)
Comment (dgilman) [reply databases/postgresql10/Portfile:4]: I wrote this mirroring what the Python group portfiles are doing. They still have the runtime dependency to pull in their stuff without the select group.

Agreed about the select files in files/

**Comment by reneeotten** [comment]
@jmroot is this how you envisioned the select stuff to be? If so, let's merge this so that the `postgresql17` PR can also be merged!

**Comment by jmroot** [comment]
`platforms any` is set twice, but otherwise looks reasonable.

**Pull request merged**